packages/react-checkbox/Spec.md

```diff
@@ -2,62 +2,260 @@
+      className={state.inputClassName}
+      ref={state.inputRef}
+      id={state.inputId}
+      onChange={state.inputOnChange}
+    />
+  </div>
+  {state.labelPosition === 'end' && <slots.label {...slotProps.label} />}
+</slots.root>
+```
 
-## Behaviors
+### **DOM**
+
+In this case, the label position is set to end as per default.
+
+```tsx
+<div {/* Root Element */}>
+  <div {/* Checkbox */}>
+    <div {/* Icon */}>
+      ...Icon
+    </div>
+    <input id="cbox-1" type="checkbox" aria-checked="..." aria-label="..." aria-labelledby="..."
+      aria-disabled="..." aria-posinset="..." aria-setsize="..." />
+  </div>
+  <label for="cbox-1">
+    Example Checkbox
+  </label>
+</div>
+```
 
-_Explain how the component will behave in use, including:_
+## Migration
+
+See [MIGRATION.md](MIGRATION.md)
+
+## Behaviors
 
-- _Component States_
-- _Interaction_
-  - _Keyboard_
-  - _Cursor_
-  - _Touch_
-  - _Screen readers_
+- Keyboard
+  - When checkbox is focused, if `space` is pressed the checkbox's state is toggled.
+- Cursor
+  - When clicked the checkbox's state is toggled.
+- Focus
+  - A checkbox can be focused to interact with it using `space`.
+  - In case of a label having a link or information button, items inside the label may be focused.
 
 ## Accessibility
 
-Base accessibility information is included in the design document. After the spec is filled and review, outcomes from it need to be communicated to design and incorporated in the design document.
-
-- Decide whether to use **native element** or folow **ARIA** and provide reasons
-- Identify the **[ARIA](https://www.w3.org/TR/wai-aria-practices-1.2/) pattern** and, if the component is listed there, follow its specification as possible.
-- Identify accessibility **variants**, the `role` ([ARIA roles](https://www.w3.org/TR/wai-aria-1.1/#role_definitions)) of the component, its `slots` and `aria-*` props.
-- Describe the **keyboard navigation**: Tab Oder and Arrow Key Navigation. Describe any other keyboard **shortcuts** used
-- Specify texts for **state change announcements** - [ARIA live regions
-  ](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) (number of available items in dropdown, error messages, confirmations, ...)
-- Identify UI parts that appear on **hover or focus** and specify keyboard and screen reader interaction with them
-- List cases when **focus** needs to be **trapped** in sections of the UI (for dialogs and popups or for hierarchical navigation)
-- List cases when **focus** needs to be **moved programatically** (if parts of the UI are appearing/disappearing or other cases)
+- Aria design pattern: [Checkbox](https://www.w3.org/TR/wai-aria-practices-1.2/#checkbox).
+- If a label is used with the checkbox, an `aria-label` will be set on the `input` element.
+- Checkbox will have a tri-state `aria-checked`.
+- Elements inside the checkbox's label can be focused if there are elements such as links and info buttons.
```

Comment:
There's no special handling in checkbox for focusable elements in the label right? This is just noting the standard behavior if a focusable item is in the label?

Reply:
What I was meaning is in case there's a link inside the label or a focusable item, then they should be allowed to be focused. The label itself cannot be focused. (We discussed this offline but I'm adding it for reference :) )